### ISSUE TITLE
[tests] Add tests based on `ActivitySourceTest` for all missing components

### ISSUE THREAD

**Issue by radical** [dotnet/aspire#3014]
Eric: All the components that have tracing should have these tests based on `ActivitySourceTest`, like: https://github.com/dotnet/aspire/blob/54485cfa8f77da6b6d3fb02b3fd63f31dc3cf8ff/tests/Aspire.Npgsql.Tests/ConformanceTests.cs#L119-L133

TODO:
- [ ] list of components that are missing these

cc @eerhardt

**Comment by davidfowl** [comment]
Why? Are we doing bad things with activity listeners or is otel?

**Comment by radical** [comment]
> Why? Are we doing bad things with activity listeners or is otel?

I'm not sure I understand. These tests need to be added in the concrete test classes, and need to be run with `RemoteExecutor` because IIUC, the activity listeners are global for the process and get events from other tests too. Please correct me if I'm wrong, @eerhardt .

**Comment by davidfowl** [comment]
Have we done this?

**Comment by radical** [comment]
Yes, this looks done AFAICS. I tried looking at the `public .*Tracing.*Activity` tests, and all of them are using `ActivitySourceTest`.
